daemon/daemon.go

```diff
@@ -358,10 +354,13 @@ func (d *Daemon) Start() {
 		case state.RestartDaemon:
 			d.tomb.Kill(nil)
 		case state.RestartSystem:
-			cmd := exec.Command("shutdown", "+10", "-r", shutdownMsg)
-			if out, err := cmd.CombinedOutput(); err != nil {
-				logger.Noticef("%s", osutil.OutputErr(out, err))
+			bootloader, err := partition.FindBootloader()
+			if err != nil {
+				logger.Noticef("cannot get bootloader: %s", err)
+				break
 			}
+			// Reboot in 10 minutes
+			bootloader.RebootForUpdate(10)
```

Comment:
I would prefer it bootloader.RebootForUpdate() takes a time.Time - this way it is more clear what the parameter means and the code is easier to read (even without a comment).